lib/ansible/plugins/action/uri.py

```diff
@@ -46,6 +46,11 @@ def run(self, tmp=None, task_vars=None):
             self._transfer_file(src, tmp_src)
             self._fixup_perms2((self._connection._shell.tmpdir, tmp_src))
 
+            try:
+                decrypt = boolean(self._task.args.get('decrypt', True), strict=False)
+                tmp_src = self._loader.get_real_file(tmp_src, decrypt=decrypt)
+            except AnsibleFileNotFound as e:
+                raise AnsibleActionFail("could not find src=%s, %s" % (tmp_src, to_text(e)))
```

Comment:
like in copy and other action plugins that support this, you should add a toggle for the user to be able to actually post a vaulted file if that is what they want.

Reply:
Will do that and send the PR once done

Reply:
![Screenshot from 2019-09-01 14-14-13](https://user-images.githubusercontent.com/12994292/64073972-e2110a00-ccc2-11e9-8103-64bf0ea3f26f.png)

Reply:
Decrypt default is set to True